Sources/hostmgr/commands/generate/GenerateBuildkiteJobScript.swift

```diff
@@ -11,101 +11,34 @@ struct GenerateBuildkiteJobScript: ParsableCommand {
     )
 
     func run() throws {
-        let path = try FileManager.default.createTemporaryFile(containing: generateBuildScript()).path
-        print(path)
-    }
-
-    func generateBuildScript() throws -> String {
-        let exports = try generateExports()
-            .map { "export \($0.key)=\"\($0.value)\"" }
-            .joined(separator: "\n")
-
-        return [
-            "source ~/.circ",               // Need to source .circ first in order to set up the SSH session properly
-            exports,                        // Declare all of our environment variables
-            "buildkite-agent bootstrap"    // Then let's go!
-        ].joined(separator: "\n")
-    }
+        var scriptBuilder = BuildkiteScriptBuilder()
 
-    // swiftlint:disable:next function_body_length
-    func generateExports(
-        from environment: [String: String] = ProcessInfo.processInfo.environment
-    ) throws -> [String: String] {
+        scriptBuilder.copyEnvironmentVariables(prefixedBy: "BUILDKITE_")
+        scriptBuilder.addDependency(atPath: "~/.circ")
+        scriptBuilder.addCommand("buildkite-agent bootstrap")
```

Comment:
💄  Nitpick: Reorder those lines to match the order in which those various sections will be output during `scriptBuilder.build()`

```suggestion
        scriptBuilder.addDependency(atPath: "~/.circ")
        scriptBuilder.copyEnvironmentVariables(prefixedBy: "BUILDKITE_")
        scriptBuilder.addCommand("buildkite-agent bootstrap")
```